src/DBmysql.php

```diff
@@ -402,17 +402,16 @@ public function doQuery($query)
             }
         }
 
+        $this->last_query_warnings = $this->fetchQueryWarnings();
         if ($is_debug && $CFG_GLPI["debug_sql"]) {
             $TIME = (float) $TIMER->getTime();
             $debug_data['time'] = (int) ($TIME * 1000);
             $debug_data['rows'] = $this->affectedRows();
             $DEBUG_SQL["times"][$SQL_TOTAL_REQUEST] = $TIME;
             $DEBUG_SQL['rows'][$SQL_TOTAL_REQUEST] = $this->affectedRows();
+            $DEBUG_SQL['warnings'][$SQL_TOTAL_REQUEST] = $this->last_query_warnings;
```

Comment:
It seems `fetchQueryWarnings()` run a `SHOW WARNINGS` query.

If it is executed before `$debug_data['rows'] = $this->affectedRows()`, wouldn't the results of `SHOW WARNINGS` replace the rows returned by the previous query ?

(I didn't test it, just wondering).

Reply:
No idea; that could indeed be a problem. Original code can be restored, and a condition added for populating the `$DEBUG_SQL['warnings'][$SQL_TOTAL_REQUEST]` part.

Reply:
Seems not impacting, as I see the affected rows count. Values are the same, no matter the order of the two lines, but I'll change this to make the code safer.